### PR TITLE
fix(messaging): Throw exception if APNS token is not yet available

### PR DESCRIPTION
**Pull request by russellwheatley**
## Description

The Apple event handler that receives the APNS token may not have been called before the User starts to make API requests such as `getToken()` & `subscribeToTopic()`. We have set an exception when it isn't set and also updated documentation to reflect the requirement.

## Related Issues

closes https://github.com/firebase/flutterfire/issues/10625
closes https://github.com/firebase/flutterfire/issues/10656

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [ ] I read and followed the [Flutter Style Guide].
- [ ] I signed the [CLA].
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [X] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/firebase/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Review by Lyokone — APPROVED** [review]

**Review by exaby73 — APPROVED** [review]

File: docs/cloud-messaging/client.md [hunk 1]
```diff
@@ -144,8 +144,20 @@ To retrieve the current registration token for an app instance, call
 ask the user for notification permissions. Otherwise, it returns a token or
 rejects the future due to an error.
 
+Warning: From iOS SDK 10.4.0 and higher, it is a requirement that the APNS token
```
Comment (russellwheatley) [inline docs/cloud-messaging/client.md:147]: hey @kevinthecheung - documentation update that could do with a review, please 😄

Comment (kevinthecheung) [inline docs/cloud-messaging/client.md:147]:
```suggestion
Warning: In iOS SDK 10.4.0 and higher, it is a requirement that the APNs token
```

File: docs/cloud-messaging/client.md [hunk 2]
```diff
@@ -144,8 +144,20 @@ To retrieve the current registration token for an app instance, call
...
+is available before making API requests. The APNS token is not guaranteed to have been received
```
Comment (kevinthecheung) [inline docs/cloud-messaging/client.md:148]:
```suggestion
is available before making API requests. The APNs token is not guaranteed to have been received
```

**Review by kevinthecheung — APPROVED** [review]

**Comment by msarkrish** [comment]
@russellwheatley I have one doubt. I'm getting this exception, While calling subscribeToTopic method in iOS. 
My question is it mandatory to make call `getAPNSToken()` method before calling subscribeToTopic method?

Or What I have to do to avoid this exception?

**Pull request merged**